src/main/java/org/prebid/server/auction/CurrencyConversionService.java

```diff
@@ -0,0 +1,198 @@
+package org.prebid.server.auction;
+
+import io.vertx.core.Vertx;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpClient;
+import io.vertx.core.http.HttpClientResponse;
+import io.vertx.core.json.Json;
+import org.apache.commons.collections4.MapUtils;
+import org.prebid.server.currency.proto.response.CurrencyConversionRates;
+import org.prebid.server.exception.PreBidException;
+import org.prebid.server.util.HttpUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Service for price currency conversion between currencies.
+ */
+public class CurrencyConversionService {
+
+    private static final Logger logger = LoggerFactory.getLogger(CurrencyConversionService.class);
+
+    private static final String DEFAULT_BID_CURRENCY = "USD";
+    private final String currencyServerUrl;
+    private final HttpClient httpClient;
+
+    private Map<String, Map<String, BigDecimal>> latestCurrencyRates;
```

Comment:
Although this is default behavior anyway initializing this field with null will univocally warn the reader that it could bear `null` value initially and onwards and use it discreetly.